src/Mod/Sketcher/Gui/CommandCreateGeo.cpp

```diff
@@ -190,6 +190,30 @@ void removeRedundantHorizontalVertical(Sketcher::SketchObject* psketch,
     }
 }
 
+bool distanceXYorPointOnObject(bool distanceXZeroYOne, int geoId, Sketcher::PointPos posId, double distance, App::DocumentObject* obj) {

```

Comment:
In the current coding, it is ok to have such a function, but the naming could be better.

"distanceXYorPointOnObject" does not say what the function does. It could be a function that returns true if there is a distanceXY constraint on an object, and false if there is a point on object constraint. I know it is not, but only because I have read the body of the function. That is not optimal.

"FixToAxis" could be an alternative, specially if the first parameter is renamed to something more meaningful (like an enum allowing to refer to an axis).

Again, do not change it. This is just a exchange of information.

This function has some tricky parts. 
- The comparison (distance == 0), 0 is an literal integer instead of a literal double. 
- I am unsure at this moment if you want to compare with exactly 0, or if you would like to have something like < Precision::Confusion. Double values are generally not compared for equality.
- Return values, if booleans, should be true and false instead of numbers. However, it looks like you do not mean to return a boolean, but an enum like value. The client code will not be clear as "true" and "false" do not correlate well with "onAxis" and "at a distance from an Axis".

Again. Just a exchange of information. Do not change anything at this point.

Reply:
in (distance == 0) distance is the value that the user typed in the spinbox. So he either typed 0 exactly or something like 0.00024. But if he typed 0.00024 he probably needs it. So this is why I used == 0. What do you think?

Return value was just a way to get a (currently unused) feedback to know if the distance has been replaced or not. But it isn't used so we can return void. I just thought this function could be used somewhere down the line and that at that moment it could be interesting to know if it replaced or not the constraint.

Reply:
The comparison, save for comparing to a double, is then ok to be equality.

Let's adhere to the YAGNI philosophy for the return type, just for now.

Do not change anything at this point.